### PR TITLE
Coverage for several issues, which will be fixed in 3.8.5 via backporting

### PR DESCRIPTION
**Pull request by fedinskiy**
### Summary

The issues:
- https://github.com/quarkusio/quarkus/pull/40718
- https://github.com/quarkusio/quarkus/issues/37323
- https://github.com/quarkusio/quarkus/issues/40307

Please select the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Backport
- [x] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

**Comment by rsvoboda** [comment]
This mono commit doesn't follow items checked in description:
Screenshot 2024-06-05 at 11 24 15

 - There are 3 topics handled by this PR, there should be 3 commits.
 - Commit message doesn't give any idea about the concrete scenarios, I don't really care the stuff get there because of 3.8.5 release, commit message should cover highlight of the change
 - I miss explanatory failure messages for the persona facing the test failure
 - Test methods/classes could have a comment or JavaDoc entry to give a bit of context

File: security/basic/src/main/java/io/quarkus/ts/openshift/security/basic/SecurityFilter.java
Comment (mocenas) [inline security/basic/src/main/java/io/quarkus/ts/openshift/security/basic/SecurityFilter.java:14]: What is a purpose of this? I don't understand it. 
If I understood the issue https://github.com/quarkusio/quarkus/issues/40307 correctly, then this is just a workaround for issue that is fixed in 3.10.1. Is that correct?
If yes, do we really want to have a workaround in our TS? At least, it should have a comment linking the issue, fix and explaining it.
WDYT @michalvavrik ?

**Review by mocenas — CHANGES_REQUESTED** [review]
I'm not sure about the security changes.

File: security/basic/src/main/java/io/quarkus/ts/openshift/security/basic/SecurityFilter.java
Comment (michalvavrik) [reply security/basic/src/main/java/io/quarkus/ts/openshift/security/basic/SecurityFilter.java:14]: it never had anything to do with websockets. https://github.com/quarkusio/quarkus/issues/40307 is about existence of `@RouteFilter(404)`. You can have it empty and it will still be a reproducer. What you want to do is to have Reactive Route filter as then, Reactive Routes activates CDI request context.

File: security/basic/pom.xml
Comment (michalvavrik) [reply security/basic/pom.xml:16]: it should be `quarkus-rest` (no reactive in name); I think Fedor just want to avoid changes in backports :-) I would prefer to avoid `quarkus-resteasy-reactive` in main branch, no other module uses it. But I am not a reviewer here, so I'll leave you to decide.

File: security/basic/src/main/java/io/quarkus/ts/openshift/security/basic/SecurityFilter.java
Comment (fedinskiy) [reply security/basic/src/main/java/io/quarkus/ts/openshift/security/basic/SecurityFilter.java:14]: As Michal noted correctly, this is not a workaround, but condition for an issue to be reproduced. The condition, on the other hand is a workaround for Websockets, but that is another story :)

I will try to clean the filter from everything that is not required for the bug to resurface, though

Comment (fedinskiy) [reply security/basic/src/main/java/io/quarkus/ts/openshift/security/basic/SecurityFilter.java]: Everything related to websockets was removed from the filter

File: security/basic/pom.xml
Comment (mocenas) [reply security/basic/pom.xml:16]: Still IMHO the issue should be linked here, or put some comment here, why it needs to be reactive. Otherwise someone will be very confused some day, when changing something here.

File: security/basic/src/main/java/io/quarkus/ts/openshift/security/basic/SecurityFilter.java
Comment (mocenas) [inline security/basic/src/main/java/io/quarkus/ts/openshift/security/basic/SecurityFilter.java:9]: Please add explanation, why is the filter here (ideally link the issue + comment that issue only appears if security filter is here).
There is a piece of code, that does nothing on a first glance. It would be really good to have an explanation why is it necessary.

**Review by mocenas — CHANGES_REQUESTED** [review]
Please add comments/link about the security issue.
Otherwise it looks good.

**Comment by rsvoboda** [comment]
@mocenas waiting for feedback from @michalvavrik before merging?

**Comment by mocenas** [comment]
> @mocenas waiting for feedback from @michalvavrik before merging?

I don't think @michalvavrik was requested for feedback any more. IMHO we can merge it now.

**Comment by michalvavrik** [comment]
@fedinskiy can you follow-up after this PR is merged with change of `quarkus-resteasy-reactive` to `quarkus-rest`? This makes it only place were we need relocation, which is not a problem for CI, but I normally don't do them. There is no reason.

**Pull request merged**
